### ISSUE TITLE
has no member named 'packetExtended'

### ISSUE THREAD

**Issue by victorlysak** [adafruit/Adafruit_MCP2515#5]
Trying Adafruit mcp2515 example: MCP2515_CAN_Receiver

Arduino compiler shows msg:

Arduino: 1.8.19 (Linux), Board: "Adafruit Feather RP2040 CAN, 8MB (no FS), 133 MHz, Small (-Os) (standard), Disabled, Disabled, Disabled, Disabled, None, Pico SDK, IPv4 Only, W25Q080 QSPI /2, Default (UF2)"

In file included from /home/vic/Arduino/libraries/AA_MCP2515/src/CANController.h:5,
                 from /home/vic/Arduino/libraries/Adafruit_MCP2515/src/Adafruit_MCP2515.h:11,
                 from /home/vic/Arduino/libraries/Adafruit_MCP2515/examples/MCP2515_CAN_Receiver/MCP2515_CAN_Receiver.ino:5:
/home/vic/Arduino/libraries/AA_MCP2515/src/CANConfig.h:23:18: error: cannot declare field 'CANConfig::spi' to be of abstract type 'arduino::HardwareSPI'
   23 |   const SPIClass spi;
      |                  ^~~
In file included from /home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/HardwareSPI.h:2,
                 from /home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/libraries/SPI/src/SPI.h:24,
                 from /home/vic/Arduino/libraries/Adafruit_BusIO/Adafruit_SPIDevice.h:9,
                 from /home/vic/Arduino/libraries/Adafruit_MCP2515/src/Adafruit_MCP2515.h:10:
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:104:7: note:   because the following virtual functions are pure within 'arduino::HardwareSPI':
  104 | class HardwareSPI
      |       ^~~~~~~~~~~
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:109:21: note:     'virtual uint8_t arduino::HardwareSPI::transfer(uint8_t)'
  109 |     virtual uint8_t transfer(uint8_t data) = 0;
      |                     ^~~~~~~~
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:110:22: note:     'virtual uint16_t arduino::HardwareSPI::transfer16(uint16_t)'
  110 |     virtual uint16_t transfer16(uint16_t data) = 0;
      |                      ^~~~~~~~~~
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:111:18: note:     'virtual void arduino::HardwareSPI::transfer(void*, size_t)'
  111 |     virtual void transfer(void *buf, size_t count) = 0;
      |                  ^~~~~~~~
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:127:18: note:     'virtual void arduino::HardwareSPI::usingInterrupt(int)'
  127 |     virtual void usingInterrupt(int interruptNumber) = 0;
      |                  ^~~~~~~~~~~~~~
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:128:18: note:     'virtual void arduino::HardwareSPI::notUsingInterrupt(int)'
  128 |     virtual void notUsingInterrupt(int interruptNumber) = 0;
      |                  ^~~~~~~~~~~~~~~~~
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:129:18: note:     'virtual void arduino::HardwareSPI::beginTransaction(arduino::SPISettings)'
  129 |     virtual void beginTransaction(SPISettings settings) = 0;
      |                  ^~~~~~~~~~~~~~~~
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:130:18: note:     'virtual void arduino::HardwareSPI::endTransaction()'
  130 |     virtual void endTransaction(void) = 0;
      |                  ^~~~~~~~~~~~~~
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:133:18: note:     'virtual void arduino::HardwareSPI::attachInterrupt()'
  133 |     virtual void attachInterrupt() = 0;
      |                  ^~~~~~~~~~~~~~~
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:134:18: note:     'virtual void arduino::HardwareSPI::detachInterrupt()'
  134 |     virtual void detachInterrupt() = 0;
      |                  ^~~~~~~~~~~~~~~
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:136:18: note:     'virtual void arduino::HardwareSPI::begin()'
  136 |     virtual void begin() = 0;
      |                  ^~~~~
/home/vic/.arduino15/packages/rp2040/hardware/rp2040/3.6.2/cores/rp2040/api/../../../ArduinoCore-API/api/HardwareSPI.h:137:18: note:     'virtual void arduino::HardwareSPI::end()'
  137 |     virtual void end() = 0;
      |                  ^~~
/home/vic/Arduino/libraries/AA_MCP2515/src/CANConfig.h:12:95: error: cannot allocate an object of abstract type 'arduino::HardwareSPI'
   12 |   CANConfig(CANBitrate::Config bitrate_config, uint8_t cs_pin, int8_t int_pin, SPIClass spi = SPI, uint32_t spi_hz = DEFAULT_SPI_HZ) :
      |                                                                                               ^~~
/home/vic/Arduino/libraries/AA_MCP2515/src/CANConfig.h:12:89: error: cannot declare parameter 'spi' to be of abstract type 'arduino::HardwareSPI'
   12 |   CANConfig(CANBitrate::Config bitrate_config, uint8_t cs_pin, int8_t int_pin, SPIClass spi = SPI, uint32_t spi_hz = DEFAULT_SPI_HZ) :
      |                                                                                ~~~~~~~~~^~~~~~~~~
In file included from /home/vic/Arduino/libraries/AA_MCP2515/src/CANController.h:7:
/home/vic/Arduino/libraries/AA_MCP2515/src/CANIO.h:23:20: error: cannot declare field 'CANIO::_spi' to be of abstract type 'arduino::HardwareSPI'
   23 |     const SPIClass _spi;
      |                    ^~~~
/home/vic/Arduino/libraries/Adafruit_MCP2515/src/Adafruit_MCP2515.h:15:52: error: expected class-name before '{' token
   15 | class Adafruit_MCP2515 : public CANControllerClass {
      |                                                    ^
/home/vic/Arduino/libraries/Adafruit_MCP2515/src/Adafruit_MCP2515.h:31:9: error: 'CANControllerClass' has not been declared
   31 |   using CANControllerClass::filter;
      |         ^~~~~~~~~~~~~~~~~~
/home/vic/Arduino/libraries/Adafruit_MCP2515/src/Adafruit_MCP2515.h:33:9: error: 'CANControllerClass' has not been declared
   33 |   using CANControllerClass::filterExtended;
      |         ^~~~~~~~~~~~~~~~~~
/home/vic/Arduino/libraries/Adafruit_MCP2515/examples/MCP2515_CAN_Receiver/MCP2515_CAN_Receiver.ino: In function 'void loop()':
MCP2515_CAN_Receiver:56:13: error: 'class Adafruit_MCP2515' has no member named 'packetExtended'
   56 |     if (mcp.packetExtended()) {
      |             ^~~~~~~~~~~~~~
MCP2515_CAN_Receiver:60:13: error: 'class Adafruit_MCP2515' has no member named 'packetRtr'
   60 |     if (mcp.packetRtr()) {
      |             ^~~~~~~~~
MCP2515_CAN_Receiver:66:22: error: 'class Adafruit_MCP2515' has no member named 'packetId'
   66 |     Serial.print(mcp.packetId(), HEX);
      |                      ^~~~~~~~
MCP2515_CAN_Receiver:68:13: error: 'class Adafruit_MCP2515' has no member named 'packetRtr'
   68 |     if (mcp.packetRtr()) {
      |             ^~~~~~~~~
MCP2515_CAN_Receiver:70:26: error: 'class Adafruit_MCP2515' has no member named 'packetDlc'
   70 |       Serial.println(mcp.packetDlc());
      |                          ^~~~~~~~~
MCP2515_CAN_Receiver:76:18: error: 'class Adafruit_MCP2515' has no member named 'available'
   76 |       while (mcp.available()) {
      |                  ^~~~~~~~~
MCP2515_CAN_Receiver:77:32: error: 'class Adafruit_MCP2515' has no member named 'read'
   77 |         Serial.print((char)mcp.read());
      |                                ^~~~
Multiple libraries were found for "CANController.h"
 Used: /home/vic/Arduino/libraries/AA_MCP2515
 Not used: /home/vic/Arduino/libraries/CAN_Adafruit_Fork
 Not used: /home/vic/Arduino/libraries/Adafruit_CAN
 Not used: /home/vic/Arduino/libraries/CAN
exit status 1
'class Adafruit_MCP2515' has no member named 'packetExtended'

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Comment by caternuson** [comment]
Looks like wrong CAN library is being used:
```
In file included from /home/vic/Arduino/libraries/AA_MCP2515/src/CANController.h:5,
from /home/vic/Arduino/libraries/Adafruit_MCP2515/src/Adafruit_MCP2515.h:11,
from /home/vic/Arduino/libraries/Adafruit_MCP2515/examples/MCP2515_CAN_Receiver/MCP2515_CAN_Receiver.ino:5:
```

Where did the `AA_MCP2515` library come from?

The correct one is here:
https://github.com/adafruit/Adafruit_CAN

**Comment by victorlysak** [comment]
Downloaded in the lib manager in the Arduino ide, Arch linux

Also, the link, you gave, doesn't have any libraries. And as I see it, it is for some M4, not for my Adafruit RP2040 Can Bus ?
Will it work?

**Comment by caternuson** [comment]
The link is to the library repo source code. This library has that library as a dependency:
https://github.com/adafruit/Adafruit_MCP2515/blob/f283438d8021440568b3f2a8e49eccc781ff3473/library.properties#L10
It reuses the `CANController` class from that library.

If you install this library using the Arduino Library Manager, the `Adafruit_CAN` library should have been installed as a dependency.

It is unknown what the `AA_MCP2515` library is, but it should be removed. Or otherwise not used in conjunction with this library.

**Comment by victorlysak** [comment]
thank you very much for your reply

I got the board in order to make cornering lights for my car.

And now I am trying to connect to the obd2 can wires 6,14 and see if I can get some msgs. so far zero result.
tried even with that crap, python or curcuitpit.., in mu.appimage but the terminal there is dead, on thonny terminal is ok but the examples not working , and python is 3x slower, so i'll try to stay away from it.

so far this board looks crap, but i'll try to see if anything will work with it

I simply downloaded ard ide, and installed all the libraries that have adafruit ... can bus and all others that have can bus in their names.
why not? i've got 1tb space on my drive.

thank you anyway.
